### PR TITLE
Removal of Mirror Maker 1 support

### PR DESCRIPTION
**Pull request by scholzj**
This proposal handles the timeline and approach for removing the support for Mirror Maker 1 (planned to be removed in Apache Kafka 4.0).

### CODE REVIEW

File: 078-removal-of-mirror-maker-1.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
+# Removal of Mirror Maker 1
+
+Mirror Maker 1 is planned to be removed in Apache Kafka 4.0.
+This proposal discusses its removal from Strimzi.
+
+## Current situation
+
+Apache Kafka currently supports two versions of Mirror Maker.
+Mirror Maker 1, which is the original Mirror Maker version.
+And Mirror Maker 2 that is a newer replacement for Mirror Maker 2 with new architecture and features.
```
Comment (im-konge) [inline 078-removal-of-mirror-maker-1.md:10]:
```suggestion
And Mirror Maker 2 that is a newer replacement for Mirror Maker 1 with new architecture and features.
```

File: 078-removal-of-mirror-maker-1.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+* Remove the Mirror Maker 1 docs apart from a simple note that Mirror Maker 1 is not supported anymore
+* Remove `KafkaMirrorMakerAssemblyOperator`, `KafkaMirrorMakerCluster` and other related classes from the `cluster-operator` module
+
+All these changes are expected to be done before the Strimzi 0.4z release.
+
+### Warnings
+
+Mirror Maker 1 is already deprecated in Strimzi.
+Warnings about its upcoming removal (in logs as well as in the `KafkaMirrorMaker` resource conditions) are already being added in Strimzi 0.43.
+When we know the exact strimzi version where the support for Mirror Maker 1 to be removed, we will update the warning messages.
```
Comment (im-konge) [inline 078-removal-of-mirror-maker-1.md:77]:
```suggestion
When we know the exact Strimzi version where the support for Mirror Maker 1 to be removed, we will update the warning messages.
```

**Review by im-konge — APPROVED** [review]
Just small nits, the approach looks good to me. Thanks :)

**Review by PaulRMellor — APPROVED** [review]
The proposal and removal schedule makes sense

File: 078-removal-of-mirror-maker-1.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+If the Mirror Maker 1 cluster is not migrated before upgrade to Strimzi 0.4z or later, Strimzi will stop operating it.
+But it will keep it running and not delete it.
+When the user is done with the migration, they can simply delete the `KafkaMirrorMaker` custom resource and that will also delete the old Mirror Maker 1 cluster and all its resources through Kubernetes garbage collection.
+
+Users will be also responsible for deleting the `KafkaMirrorMaker` CRD once they upgrade to Strimzi 0.4z or later (and once they migrate all of their remaining clusters, as the CRD deletion would also delete the custom resources and the related Deployments/Pods).
+
+### Strimzi changes
+
+The following changes will be done to Strimzi to remove support for Mirror Maker 1:
+* Remove the `KafkaMirrorMaker` CRD from Strimzi installation files, Helm Chart and OperatorHub entires
```
Comment (ppatierno) [inline 078-removal-of-mirror-maker-1.md:64]:
```suggestion
* Remove the `KafkaMirrorMaker` CRD from Strimzi installation files, Helm Chart and OperatorHub entries
```

File: 078-removal-of-mirror-maker-1.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+Mirror Maker 1 is planned to be removed in Apache Kafka 4.0.
+This proposal discusses its removal from Strimzi.
+
+## Current situation
+
+Apache Kafka currently supports two versions of Mirror Maker.
+Mirror Maker 1, which is the original Mirror Maker version.
+And Mirror Maker 2 that is a newer replacement for Mirror Maker 1 with new architecture and features.
+Strimzi currently support both Mirror Maker versions
```
Comment (ppatierno) [inline 078-removal-of-mirror-maker-1.md:11]:
```suggestion
Strimzi currently supports both Mirror Maker versions
```

**Review by ppatierno — APPROVED** [review]
LGTM. Just a couple of typos.

File: 079-removal-of-mirror-maker-1.md [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+| 0.4x    | 3.8.x, 3.9.x             | ZooKeeper, KRaft         | no               | yes         |
+| 0.4y    | 3.8.x, 3.9.x             | ZooKeeper, KRaft         | **yes**          | yes         |
+| 0.4z    | 3.9.x, 4.0.x             | KRaft                    | no               | no          |
+| ...     | 4.0.x, 4.1.x             | KRaft                    | no               | no          |
+
+Users will be responsible for migrating their Mirror Maker 1 clusters to Mirror Maker 2 (or any other alternative they might choose) before upgrading to Strimzi 0.4z.
+While it is possible to configure Mirror Maker 2 to closely resemble the functionality of Mirror Maker 1, this proposal does not cover the detailed steps of how users should migrate from Mirror Maker 1 to Mirror Maker 2.
+
+If the Mirror Maker 1 cluster is not migrated before upgrade to Strimzi 0.4z or later, Strimzi will stop operating it.
+But it will keep it running and not delete it.
```
Comment (tombentley) [inline 079-removal-of-mirror-maker-1.md:56]: I think this sentence could be clearer. There are two 'it's which mean different things. I think we're saying "Strimzi will stop managing (but not delete) the MM1 CR or `Deployment`, so the MM1 Pod will continue to run."

**Review by tombentley — APPROVED** [review]
Thanks @scholzj

File: 079-removal-of-mirror-maker-1.md (hunk 5 above)
Comment (scholzj) [reply 079-removal-of-mirror-maker-1.md]: You are right, this is a bit confusing. I reworded it to make it (hopefully) more clear.

**Comment by scholzj** [comment]
This now has 4 binding and one non-binding +1 vote. If you plan to review this proposal, please do so by Monday EOB (CEST timezone). Otherwise, I will close the vote.

**Review by Frawless — APPROVED** [review]

**Comment by scholzj** [comment]
Approved with 5 binding and 1 non-binding +1 votes.

**Pull request merged**